docs/topics/api/abuse.rst

```diff
@@ -0,0 +1,59 @@
+=============
+Abuse Reports
+=============
+
+The following API endpoint covers abuse reporting
+
+---------------------------------
+Submitting an add-on abuse report
+---------------------------------
+
+.. _`addonabusereport-create`:
+
+The following API endpoint allows an abuse report to be submitted for an Add-on,
+either listed on https://addons.mozilla.org or not.
+Authentication is not required, but is recommended so reports can be responded
+to if nessecary.
+
+.. http:post:: /api/v3/abuse/reportaddon/
+
+    .. _addonabusereport-create-request:
+
+    :<json string addon: The add-on id, slug, or guid of the add-on (required).
+    :<json string message: The body/content of the abuse report (required).
+    :>json object|null reporter: The user who submitted the report, if authenticated.
+    :>json int reporter.id: The id of the user who submitted the report.
+    :>json string reporter.name: The name of the user who submitted the report.
```

Comment:
Is this stuff checked against the AMO users DB or is it just submitted freeform and we're expected to use the right data? If it's checked it seems we should really only need to submit a user id, username, or email and AMO can fetch the rest of this. Seems silly to submit it all.